src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/CatalogPromotion.xml

```diff
@@ -147,13 +168,62 @@ Example configuration for `percentage_discount` action type:
                     </attribute>
                 </attribute>
             </itemOperation>
+
+            <itemOperation name="admin_archive">
+                <attribute name="method">PATCH</attribute>
+                <attribute name="path">/admin/catalog-promotions/{code}/archive</attribute>
+                <attribute name="controller">Sylius\Bundle\ApiBundle\Controller\ArchiveCatalogPromotionAction</attribute>
+                <attribute name="input">false</attribute>
+                <attribute name="normalization_context">
+                    <attribute name="groups">admin:catalog_promotion:read</attribute>
+                </attribute>
+                <attribute name="openapi_context">
+                    <attribute name="summary">Archives CatalogPromotion</attribute>
+                    <attribute name="responses">
+                        <attribute name="202">
+                            <attribute name="description">CatalogPromotion resource archival request accepted</attribute>
+                        </attribute>
+                        <attribute name="400">
+                            <attribute name="description">Invalid CatalogPromotion state</attribute>
+                        </attribute>
+                        <attribute name="404">
+                            <attribute name="description">Resource not found</attribute>
+                        </attribute>
+                    </attribute>
+                </attribute>
+            </itemOperation>
+
+            <itemOperation name="admin_restore">
+                <attribute name="method">PATCH</attribute>
+                <attribute name="path">/admin/catalog-promotions/{code}/restore</attribute>
+                <attribute name="controller">Sylius\Bundle\ApiBundle\Controller\RestoreCatalogPromotionAction</attribute>
+                <attribute name="input">false</attribute>
+                <attribute name="normalization_context">
+                    <attribute name="groups">admin:catalog_promotion:read</attribute>
+                </attribute>
+                <attribute name="openapi_context">
+                    <attribute name="summary">Restores Archived CatalogPromotion</attribute>
+                    <attribute name="responses">
+                        <attribute name="202">
+                            <attribute name="description">CatalogPromotion resource restoral request accepted</attribute>
+                        </attribute>
+                        <attribute name="400">
```

Comment:
Please remove 400 and 404 error definitions, for archive and removal operations as well.

It documents automatically

![image](https://github.com/Sylius/Sylius/assets/870747/256a1949-a5da-45f3-9aae-944248c229ec)